samples/BasicDiscovery/src/main/java/basicdiscovery/BasicDiscovery.java

```diff
@@ -122,22 +132,36 @@ static void parseCommandLine(String[] args) {
 
     static CompletableFuture<Void> executeSession(MqttClientConnection connection) {
         CompletableFuture<Void> result = new CompletableFuture<>();
-        Consumer<MqttMessage> handler = (MqttMessage msg) -> {
-            String json = new String(msg.getPayload(), StandardCharsets.UTF_8);
-            
-        };
+        try {
+            Consumer<MqttMessage> handler = (MqttMessage msg) -> {
+                String json = new String(msg.getPayload(), StandardCharsets.UTF_8);
+                JsonElement root = new JsonParser().parse(json);
+                int sequence = root.getAsJsonObject().get("sequence").getAsInt();
+                if (sequence == numPublishes) {
+                    result.complete(null);
+                }
+            };
+            connection.subscribe(topic, QualityOfService.AT_LEAST_ONCE, handler).get();
```

Comment:
Qos 1 doesn’t work with greengrass.

Reply:
Thanks, I would have pulled hair out over that tomorrow